### PR TITLE
docs: Mentoring. Fixes #7307

### LINKED ISSUE #7307: Mentoring
We would like to support new contributors with mentoring. This would probably need to be lead by the main contributors:

@sarabala1979 @alexec @terrytangyuan etc

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>
Fixes #7307

First draft of mentoring process. I expect we'll change this a lot.

### CODE REVIEW

File: docs/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+* Add your name here!
+
+We'll find the right mentor for you, and let you know who to add to your PR. Then, your mentor will approve and merge your PR!
+
+## How To Become A Mentor
+
+Create a pull request which changes this page to add you to this list:
+
+Mentees:
```
Comment (terrytangyuan) [inline docs/mentoring.md:28]: Should be mentors

File: docs/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+* @alexec (Alex)
+* @simster7 (Simon)
+* @sarabala1979 (Bala)
+* @terrytangyuan (Yuan)
+
+## How To End Your Relationship
```
Comment (hblixt) [inline docs/mentoring.md:35]: Should we make it explicit that the relationship is supposed to end after the mentee's first PR is merged?

Comment (alexec) [reply docs/mentoring.md:35]: I think it can go longer.

Comment (hblixt) [reply docs/mentoring.md:35]: makes sense, we can always adjust later if it turns into a bottleneck.

**Review by hblixt — APPROVED** [review]
LGTM. I think this will be a great way of getting people started

File: docs/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Mentors and mentees must abide by the [code of conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). 
+
+## How To Get A Mentor
+
+Create a pull request which changes this page to add you to this list:
+
+Mentees:
+
+* Add your name here!
+
```
Comment (simster7) [inline docs/mentoring.md:21]: Not sure this is the best process. I suggest better creating a Github issue for each mentee and making use of the Github projects Kanban board. Each mentor would have their own columnd and mentees would each be said issue.

Comment (terrytangyuan) [reply docs/mentoring.md:21]: What kind of permissions do mentees need in order to manage the Kanban board?

Comment (alexec) [reply docs/mentoring.md:21]: I choose this to be consistent with the list of mentors, and to record relationships. But, it does mean mentees need to create a PR, which they might not be able to do so when they need a mentor. I will change it

**Comment by alexec** [comment]
@terrytangyuan @simster7 @sarabala1979 can you confirm you're happy to mentor?

**Comment by terrytangyuan** [comment]
> @terrytangyuan @simster7 @sarabala1979 can you confirm you're happy to mentor?

Yes

**Comment by sarabala1979** [comment]
> @terrytangyuan @simster7 @sarabala1979 can you confirm you're happy to mentor?

yes

**Comment by alexec** [comment]
Approval requested. There is some issue with Github and admin merge today.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**